docs/typescript.md

```diff
@@ -0,0 +1,53 @@
+# Typescript
+
+This guide is an example of how to develop a function in Typescript with
+the Functions Framework.
+
+1. Use [gts](https://github.com/google/gts) to configure Typescript.
+
+    ```sh
+    npx gts init
+    ```
+
+2. Install the required packages:
+
+    ```sh
+    npm install @google-cloud/functions-framework
+    # an HTTP signature requires express types
+    npm install @types/express --save-dev
+    # https://www.npmjs.com/package/tsc-watch
+    npm install tsc-watch --save-dev
+    ```
+
+3. Add a `start` script to `package.json`, passing in the `--source` flag to
+   point to the compiled code directory (configured by `gts` in this example).
+   Also add a `watch` script to use for development:
+
+    ```js
+      "scripts": {
+        "start": "functions-framework --source=build/src/ --target=helloWorld",
+        "watch": "tsc-watch --onSuccess 'npm start'",
```

Comment:
Unfortunately I wasn't able to get `tsc-watch` to properly refresh when changing code.

Luckily, I tried a different solution that works really well that I think we should document:

```
    "start": "functions-framework --source=build/src/ --target=helloWorld",
    "watch": "concurrently -k -p \"[{name}]\" -n \"TypeScript,Node\" -c \"yellow.bold,cyan.bold,green.bold\" \"npm run watch-ts\" \"npm run watch-node\"",
    "watch-node": "nodemon --watch ./build/ --exec npm run start",
    "watch-ts": "tsc -w"
```

This is the solution provided here: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/issues/13#issuecomment-526628422

Here is a screenshot:

![Screen Shot 2020-08-09 at 23 52 17](https://user-images.githubusercontent.com/744973/89752839-8686f900-da9b-11ea-8a3b-3a6861cd8242.png)

Was this watch working for you when changing code?

Reply:
`tsc-watch` does indeed work when code changes are made on my end that result in  successful compilation (`--onSuccess`)  🤔, do you mind trying. again?

```
    "watch": "tsc-watch --onSuccess 'npm start'",
```

Reply:
I'm not seeing the Node process properly refresh. It sometimes does, but lags, or never refreshes.

Here's a GIF where you can see me refreshing and not seeing updates:
![gif2](https://user-images.githubusercontent.com/744973/89817645-254c3d80-db0e-11ea-8b8e-b33f4a2a956f.gif)

Reply:
Oh shucks ok, FWIW I've heard `ts-node` and `nodemon` can also achieve this as per https://stackoverflow.com/questions/37979489/how-to-watch-and-reload-ts-node-when-typescript-files-change

Any preference?

Reply:
When testing, I was able to get things working like above which uses `nodemon`: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/pull/192#discussion_r467687210

I don't have a real preference so long as things work. The `nodemon` solution seems fine.

Reply:
ooc what does Google do internally for this today? Or are functions generally written in js?

Reply:
For serverless, we don't really have workflows where we need realtime updates to the FF. We don't dogfood Functions as much as I'd like. We're mostly using tsc --watch and running unit tests for the TS modules we have.

But this is important for external use of FFs and Cloud Functions / Cloud Run devX.

Reply:
Alright I've settled on a nodemon+concurrently but with just 1 script instead of 3 as per https://fireship.io/snippets/hot-reload-node-typescript-server/